### DISCUSSION TITLE
reuse component from strapi-plugin-content-manager in my plugin

### DISCUSSION THREAD

**Discussion by ha-akelius** [strapi/strapi#7078] [Legacy Topics]
**[Question]**
how can I reuse component from strapi-plugin-content-manager in my custom plugin?
I  need to reuse  strapi-plugin-content-manager/admin/src/containers/ListView component

**[What I did]**
I installed strapi-plugin-content-manager in my custom plugin
now I am getting
```
Error: ./plugins/menu/node_modules/strapi-plugin-content-manager/admin/src/containers/ListView/index.js 327:4
Module parse failed: Unexpected token (327:4)
You may need an appropriate loader to handle this file type, currently no loaders are configured to process this file. See https://webpack.js.org/concepts#loaders
| 
|   return (
>     <>
|       <ListViewProvider
|         data={data}
```

**Comment by MattieBelt** [comment]
Hi @ha-akelius,

You cannot easily use the components from the content-manager plugin, instead you could use [buffet](https://github.com/strapi/buffet) or the components from the [helper-plugin](https://github.com/strapi/strapi/tree/master/packages/strapi-helper-plugin/lib/src/components) (this helper is undocumented and not future proof). 

`<CustomList/>` [from `@buffet/custom`](https://www.buffetjs.io/storybook/?path=/story/custom--list) is best to use.

**Reply by ha-akelius** [reply]
Hi @MattieBelt 
Thank you for your reply 
I achieved what I need in a different way, not by custom plugins,

what I need to add a tree before the `ListViewProvider`, 
(so I want to reuse `content-manager plugin` to get the benefits of all data + filtering in the plugin)

this what I did, (and it is working):
the idea I followed [Update the Content Manager](https://strapi.io/documentation/v3.x/guides/custom-admin.html#update-the-content-manager)
- add: extensions/content-manager/package.json (with @material-ui/core, @material-ui/lab new dependencies)
- add: extensions/content-manager/admin/src/containers/ListViewProvider/index.js

and Inside ListViewProvider/index.js:
- I added the tree, 
- calling `props.onChangeParams` when the tree changed


but still, I think there should be a better way to change/reuse the `content-manager plugin`, 
especially when we need to use a third UI library

**Reply by MattieBelt** [reply]
I think your approach is a better one in this case, when trying to modify the plugin, extensions is the way to go.

I do agree that the used components in strapi should be more reusable.
